packages/airqo-device-analysis/airqloudanalysis/airqoairqloudanalysis/devicespecifics.py

```
@@ -0,0 +1,1034 @@
+"""### Battery Performance
+#### Color coding
+* Green -> Device is on and posted with in the hour
+* Grey -> Device is off and didn't post with in the hour"""
+# Function to calculate the average uptime of the devices
+def air_qloud_battery(df,start , end, aq_data, air_qlouds, device_names):
+    # Check if both lists are empty
+    if len(air_qlouds) > 0 and len(device_names) > 0:
+      return air_qlouds_and_device_names_error
+    if len(air_qlouds) > 0:
+      # time span
+      weekly_df, analysis_duration_days, first_date_in_df, last_date_in_df = create_dates(start, end)
```

Ruff (0.15.12)

[warning] Unpacked variable `weekly_df` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)